### PR TITLE
Upgrade gcc +

### PR DESCRIPTION
**Pull request by joewragg**
Use trixie as builder and upgrade gcc from trixie for runner

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM debian:bookworm-slim as builder
+FROM ubuntu:24.04 as builder
```
Comment (jcoupey) [inline Dockerfile:1]: We should probably stick to the initial choice of using a debian image (especially if it's smaller) if using trixie is fine.

**Review by jcoupey — COMMENTED** [review]
The build failed for some reason, not sure it is related to the change here?

File: Dockerfile (hunk 1 above)
Comment (joewragg) [reply Dockerfile]: OK I've changed back to debian trixie

**Comment by jcoupey** [comment]
The build is OK but then the test script fails here:

https://github.com/VROOM-Project/vroom-docker/blob/c24f900b2f4886ae12411ebd20670684ca2e5d23/tests/test.sh#L32-L37

**Comment by joewragg** [comment]
Just pushed a fix but it means that we are manually installing node into the debian-trixie runner image which isn't ideal at all

**Comment by joewragg** [comment]
OK latest push is to install the required packages from trixie instead

**Comment by joewragg** [comment]
Any chance we can get this merged? @jcoupey

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+*    text eol=lf
```
Comment (jcoupey) [inline .gitattributes:1]: Is this still required? Looks like it was one of the iterative changes but the actual fix is rather in the additional upgrade step in the Dockerfile.

Comment (joewragg) [reply .gitattributes]: It's not required for this fix although nice to have for windows users. I've reverted the commit so only the dockerfile changes are present in the PR now 👍

**Pull request merged**